service/controller.go

```diff
@@ -1223,8 +1226,24 @@ func (s *service) ControllerPublishVolume(
 		for key, value := range volumeContext {
 			log.Printf("    [%s]=%s", key, value)
 		}
+		// Check for AZNetwork key
+		if azNet, ok := volumeContext["AZNetwork"]; ok && azNet != "" {
+			var err error
+			log.Debugf("***[AZNETWORK] Found AZNetwork key:")
+			newExportIP, err = s.getIpsFromAZNetworkLabel(ctx, req.GetNodeId(), azNet)
+			if err != nil {
+				log.Errorf("***[AZNETWORK] Error checking AZNetwork label match: %v", err)
+				return nil, status.Error(codes.Internal, "AZNetwork label check failed")
```

Comment:
Can we include the error here?

`status.Error(codes.Internal, fmt.Sprintf("getting AZNetwork IPs: %v", err)`